src/t_stream.c

```diff
@@ -1188,10 +1170,11 @@ int streamIteratorGetID(streamIterator *si, streamID *id, int64_t *numfields) {
             serverAssert(*numfields>=0);
 
             /* If current >= start, and the entry is not marked as
-             * deleted, emit it. */
+             * deleted or tombstones are included, emit it. */
             if (!si->rev) {
                 if (memcmp(buf,si->start_key,sizeof(streamID)) >= 0 &&
-                    !(flags & STREAM_ITEM_FLAG_DELETED))
+                    (!si->skip_tombstones ||
+                    (!(flags & STREAM_ITEM_FLAG_DELETED) && si->skip_tombstones)))
```

Comment:
i think the check for skip_tombstones to be true is excessive
```suggestion
                    (!si->skip_tombstones || !(flags & STREAM_ITEM_FLAG_DELETED)))
```

Reply:
:facepalm: fixed